File: sdk/java-sdk/src/main/java/com/evernym/verity/sdk/protocols/issuersetup/v0_6/IssuerSetupImplV0_6.java
package com.evernym.verity.sdk.protocols.issuersetup.v0_6;

import com.evernym.verity.sdk.exceptions.VerityException;
import com.evernym.verity.sdk.protocols.Protocol;
import com.evernym.verity.sdk.utils.Context;
import org.json.JSONObject;

import java.io.IOException;

public class IssuerSetupImplV0_6 extends Protocol implements IssuerSetupV0_6 {

    public IssuerSetupImplV0_6() {
        super();
    }

    public void create(Context context) throws IOException, VerityException {
        send(context, createMsg(context));
    }

    @Override
    public JSONObject createMsg(Context context) {
        JSONObject message = new JSONObject();
        message.put("@type", getMessageType(CREATE));
        message.put("@id", getNewId());
        return message;
    }

    @Override
    public byte[] createMsgPacked(Context context) throws VerityException {
        return packMsg(context, createMsg(context));
    }

    @Override
    public void currentPublicIdentifier(Context context) throws IOException, VerityException {
        send(context, currentPublicIdentifierMsg(context));
    }

    @Override
    public JSONObject currentPublicIdentifierMsg(Context context) {
        JSONObject message = new JSONObject();
        message.put("@type", getMessageType(CURRENT_PUBLIC_IDENTIFIER));
        message.put("@id", getNewId());
        return message;
    }

    @Override
    public byte[] currentPublicIdentifierMsgPacked(Context context) throws VerityException {
        return packMsg(context, currentPublicIdentifierMsg(context));
    }
}

File: sdk/dotnet-sdk/src/VeritySDK/Protocols/IssuerSetup/IssuerSetupImplV0_6.cs
using System.Json;

namespace VeritySDK
{
    /*
     * NON_VISIBLE
     *
     * This is an implementation of IssuerSetupImplV0_6 but is not viable to user of Verity SDK. Created using the
     * static IssuerSetup class
     */
    public class IssuerSetupImplV0_6 : IssuerSetupV0_6
    {
        public IssuerSetupImplV0_6() { }

        public override void create(Context context)
        {
            send(context, createMsg(context));
        }

        public override JsonObject createMsg(Context context)
        {
            JsonObject message = new JsonObject();
            message.Add("@type", messageType(CREATE));
            message.Add("@id", getNewId());
            addThread(message);
            return message;
        }

        public override byte[] createMsgPacked(Context context)
        {
            return packMsg(context, createMsg(context));
        }

        public override void currentPublicIdentifier(Context context)
        {
            send(context, currentPublicIdentifierMsg(context));
        }

        public override JsonObject currentPublicIdentifierMsg(Context context)
        {
            JsonObject message = new JsonObject();
            message.Add("@type", messageType(CURRENT_PUBLIC_IDENTIFIER));
            message.Add("@id", getNewId());
            addThread(message);
            return message;
        }

        public override byte[] currentPublicIdentifierMsgPacked(Context context)
        {
            return packMsg(context, currentPublicIdentifierMsg(context));
        }
    }
}